h2o-core/src/main/java/water/init/NetworkInit.java

```diff
@@ -394,7 +395,11 @@ public static void initializeNetworkSockets( ) {
 
     // Late instantiation of Jetty object, if needed.
     if (H2O.getJetty() == null && !H2O.ARGS.disable_web) {
-      H2O.setJetty(new JettyHTTPD());
+      if(H2O.ARGS.disable_web) {
+        H2O.setJetty(new DummyServer());
```

Comment:
For the following reasons:

1. Just because I don't like nulls. DummyServer is a form of NullObject pattern.
2. Much easier fix for my original problem is also possible: there was missing H2O.ARGS.disable_web check in one of the places where it was required (method 'calcNextUniqueModelId'). My change guarantees behavior even if someone forgot checking the flag: it will not crash with NPE, it will not try to read uninitialized variables, it will try to load Jetty.